skyportal/handlers/api/internal/image_analysis.py

```diff
@@ -0,0 +1,607 @@
+    ra2: float
+        Second point RA
+    dec2: float
+        Second point Dec
+
+    Returns
+    -------
+    float
+        Distance in degrees
+
+    """
+    try:
+        x = np.sin(np.deg2rad((ra1 - ra2) / 2))
+        x *= x
+        y = np.sin(np.deg2rad((dec1 - dec2) / 2))
+        y *= y
+
+        z = np.cos(np.deg2rad((dec1 + dec2) / 2))
+        z *= z
+
+        return np.rad2deg(2 * np.arcsin(np.sqrt(x * (z - y) + y)))
+    except Exception as e:
+        raise e
+
+
+def reduce_image(
+    image,
+    header,
+    obj_id,
+    instrument_id,
+    user_id,
+    gain,
+    matching_radius,
+    catalog_name_refinement,
+    catalog_name_crossmatch,
+    template_name,
+    method,
+    detect_cosmics=False,
+):
+    """Reduce an image: Perform astrometric and photometric calibration, and extract photometry to add it to the database.
+
+    Parameters
+    ----------
+    image: numpy.ndarray
+        Image data
+    header: astropy.io.fits.Header
+        Image header
+    obj_id: str
+        Object ID
+    instrument_id: int
+        Instrument ID
+    user_id: int
+        User ID
+    detect_cosmics: bool
+        Run LACosmic on the image
+
+    Returns
+    -------
+    None
+
+    """
+
+    try:
+        workdir_sextractor_obj = tempfile.mkdtemp(prefix='sex')
+        workdir_sextractor_obj1 = tempfile.mkdtemp(prefix='sex1')
+        workdir_psfex = tempfile.mkdtemp(prefix='psfex')
+
+        with DBSession() as session:
+            instrument = session.scalars(
+                sa.select(Instrument).where(Instrument.id == instrument_id)
+            ).first()
+            user = session.scalars(sa.select(User).where(User.id == user_id)).first()
+
+            time = utils.get_obs_time(header, verbose=False)
+            filt = header.get('FILTER')
+
+            wcs = WCS(header)
+
+            # Masking saturated stars and cosmic rays
+            # The mask is a binary frame with the same size as the image where True means that this pixel should not be used for the analysis
+            mask = image > 0.9 * np.max(image)
+
+            if detect_cosmics:
+                cmask, cimage = astroscrappy.detect_cosmics(image, mask, verbose=False)
+                log(f'Done masking cosmics: {np.sum(cmask)} pixels masked')
+                mask |= cmask
+
+            # ## Detect and measure the objects
+            # The detection is based on building the noise model through (grid-based) background and background rms estimation, and then extracting the groups of connected pixels above some pre-defined threshold
+            # Detecting objects using SExtractor and getting their measurements in apertures with 3 pixels radius
+            # edge argument : control the rejection of objects detected too close to frame edge
+
+            # Astropy Table, ordered by the object brightness
+            obj = photometry.get_objects_sextractor(
+                image, gain=gain, aper=3.0, edge=10, _workdir=workdir_sextractor_obj
```

Comment:
here are some more tunable parameters. I don't know if you want to address this now, but maybe in a future PR.

Reply:
Yes, those along with (a lot more) parameters will be de-hardcoded in a future PR.

Reply:
Hi @mcoughlin. I see different hardcoded values for the aperture throughout the `reduce_image(...)` method. Any reason why:

- There are different values and not just one?
- Why this should be hardcoded?